src/js/core.js

```diff
@@ -0,0 +1,126 @@
+var d3 = require('d3');
+var _ = require('underscore');
+var utils = require('./utils');
+
+function compute_svg_width(rect_width, rect_padding, row_length) {
```

Comment:
I guess I'm undecided on this one. In general I prefer fewer functions but I see your point. In this case the code is really so simple, though, that you could almost even infer its derivation by context without a comment